src/main/java/uk/gov/hmcts/reform/prl/services/hearingmanagement/HearingManagementService.java

```diff
@@ -515,4 +507,52 @@ private EmailTemplateVars buildApplicantOrRespondentEmail(CaseData caseData, Str
                 .hearingDetailsPageLink(dashboardUrl + "/dashboard")
                 .build();
     }
+
+    public void caseNextHearingDateChangeForHearingManagement(NextHearingDateRequest nextHearingDateRequest) throws Exception {
+
+        log.info("Processing the callback for the caseId {} with next hearing date {}", nextHearingDateRequest.getCaseRef(),
+                 nextHearingDateRequest.getNextHearingDetails().getNextHearingDate());
+
+        String userToken = systemUserService.getSysUserToken();
+        String systemUpdateUserId = systemUserService.getUserId(userToken);
+        log.info("Fetching the Case details based on caseId {}", nextHearingDateRequest.getCaseRef());
+        createEventForNextHearingDate(nextHearingDateRequest, userToken, systemUpdateUserId,
+                                                                      NEXT_HEARING_DATE_UPDATE_SUCCESS);
+    }
+
+    private CaseDetails createEventForNextHearingDate(NextHearingDateRequest nextHearingDateRequest, String userToken,
+                                                      String systemUpdateUserId, String eventId) {
+
+        StartEventResponse startEventResponse = coreCaseDataApi.startEventForCaseWorker(
```

Comment:
we already have methods that contains these corecasedataapi methods. can we reuse them?

Reply:
CoreCaseDataService.java has method call triggerEvent with exact block of code, however that method not returning anything. But in our case the createEvent should return CaseDetails object.